Examples/RPC/rust/src/main.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate hyper;
+extern crate reqwest;
+extern crate protobuf;
+
+use std::env;
+use std::str::FromStr;
+use reqwest::header::ContentType;
+use reqwest::mime::Mime;
+use protobuf::parse_from_bytes;
+use protobuf::Message;
+
+mod language_service;
+
+header! { (XGoogApiKey, "X-Goog-Api-Key") => [String] }
+
+fn main() {
+    let google_api_key = env::var("GOOGLE_API_KEY").unwrap();
+
+    let mut document = language_service::Document::new();
```

Comment:
I think this might be easier to understand if it either went completely "outside to inside" or "inside to outside". The current implementation mixes them. It starts with `Document`, _then_ sets its properties (the stuff inside it), then places it into the request (the stuff outside it).

It took me a couple reads to get a handle on this despite the fact that I knew the structure. I assert that an outside-in approach is easiest to follow, even if it results in more (or uglier) code.

Reply:
I've added a few blank lines to make the structure clearer. Otherwise I kept the structure as-is... I think there's always going to be some amount of "inside to out" since we are building a data structure and then converting it to binary (the final "outside" form). But let's discuss further (outside of this PR) and possibly revisit later.